File: rstudio/source/mlflow_test_experiment.R
# Run this code to test MLflow Connection and functionality (either copy or create into user folder after login)

library(mlflow)
remote <- "http://ds-api-mlflow:5000/mlflow" # set MLflow server URI
experiment <- "/my-experiment-r"
mlflow_set_tracking_uri(remote)
mlflow_set_experiment(experiment)
mlflow_log_param("a", "1")

# - To store artifacts go to terminal and activate conda base environment:
#   conda activate base
# - Create/adjust log_artifacts.py (change experiment name and insert artifact name)
# - Create test artifact with e.g.:
#   echo "Hello World" > hello.txt
# - Run the python script:
#   python log_artifacts.py

File: jupyter/source/mlflow_test_experiment.py
# Run this code to test MLflow Connection and functionality (either copy code into notebook or run from terminal)

import mlflow
remote_server_uri = "http://ds-api-mlflow:5000/mlflow" # set MLflow server URI
mlflow.set_tracking_uri(remote_server_uri)
mlflow.set_experiment("/my-experiment-py")
with mlflow.start_run():
    mlflow.log_param("a", 1)
    mlflow.log_metric("b", 2)

# check artifact location
mlflow.get_artifact_uri()

# create test artifact
with open("hello.txt", "w") as f:
    f.write("Hello World!")

# log artifact
mlflow.log_artifact("hello.txt")